### DISCUSSION TITLE
Setting ObjectCaching to false does not seem to resolve blurry text

### DISCUSSION THREAD

**Discussion by Busata** [fabricjs/fabric.js#10258] [General]
We're using fabricjs for an internal app that renders schematics. Since some of the schematics were becoming sluggish in performance we're looking at improving it. The first step we took was upgrading the outdated version (v4.6.0) to the latest (v6.4.0). 

Before the upgrade we had set objectCaching to false. I've noticed a big increase in performance (which makes sense) when setting the objectCaching to true. (After the upgrade to 6.4.0 this is no longer necessary as afaik there is no global caching anymore and it has to be set per object). However, some texts were becoming unreadable, blurry/smudged.  I read in one of the tickets that it's possible we hit cache size limits, but while increasing this limits solves the blurryness, the performance wasn't consistent anymore. Some schematics improved, some got even slower, so I assume we're hitting limits there too. 

My current strategy was to try to *not* cache the texts in the first place. I've tried this by providing a objectCache: false to every FabricText option, but that doesn't seem to make a difference. I've then intercepted the canvas.add() call and set the property for each object we add as well, which doesn't seem to make a difference either. Is there something I'm missing about this property? Does it need to be done for everything in the group hierarchy? It's very hard to find documentation for the current version so I've been digging through tickets with no success.

![image](https://github.com/user-attachments/assets/251ffa34-9379-405b-8752-9e7173837358)

**Comment by asturur** [comment]
If you have a group the group cachin will win over the single object caching.
So text will be cached.
How many text objects do you have in a schematic? If you don't use styles or spacing render a text shouldn't be that slow.

Can you share a JSON export of a design that is sluggish?

To remove caching globally in fabric 6 please look for the default values object that is a static property on the class.
You can add objectCaching: false there to disable it for that specific class.

Documentation is behind on fabric6, yesterday i managed to move the  new website on the standard domain and now i should move faster with docs.

**Comment by Busata** [comment]
[fabric.json](https://github.com/user-attachments/files/17675435/fabric.json)

This is an export from the canvas but I've replaced the "text": "..." properties for privacy, they're usually either single characters or short descriptions (one sentence max)

I notice that it's a 20mb file so that's not a good start I reckon. Is there perhaps a way to render the same text in different locations? as I notice we sort of render the same things over and over in different locations, but equal texts are sometimes sharp, but often get blurry (due to exceeding the cache). Any tips are welcome! It was initially never intended to render these schematics so big but we ended up having to concatenate schematics in the same canvas so most of the application works fine, it's just for these larger ones that we get into performance trouble.

**Comment by heaxo** [comment]
@asturur Hello, can you help me look at this problem, I hope to solve it, or tell me the implementation idea[#10261](https://github.com/fabricjs/fabric.js/discussions/10261)
